Simple Project descriptor

jkutner opened the pull request:
[Readable](https://github.com/buildpack/rfcs/blob/project-descriptor-simple/text/0000-project-descriptor.md)

josegonzalez commented on text/0000-project-descriptor.md:
```diff
@@ -0,0 +1,281 @@
+  name = "<string>"
+
+  [[images.build.env]]
+  name = "<string>"
```
Should we add `value = "<string>"` to these two?

jkutner replied:
yes 🤦‍♂️

jkutner commented:
Updated based on discussion in working group. Changes include:

* added `value` for env vars
* changed `processes.cmd` to `processes.command` and included all other fields from `launch.toml`
* Removed `stack`
* Removed `publish`
* Removed `optional`
* Moved `license`, `author`, `source`, etc under `[metadata]`
* Added `[extensions]`

jkutner commented:
There is an unresolved question regarding how to tag multiple `[[images]]` w/o:
1. requiring developers to commit a registry URL/host to Git
1. requiring developers to manually retag the image after building

One of the options, which I'm proposing as a solution, is to have a platform use two value to construct the full image tag: `[[images]] id` and a new `--registry` flag on `pack build`. Thus, you might see a `project.toml` containing this:

```toml
[[images]]
id = "myorg/gateway"

[[images]]
id = "myorg/service1"
```

A user could then run:

```
$ pack buildpack myimage --registry gcr.io --publish
```

jkutner commented:
I've added [an example using image id prefix/suffix](https://github.com/buildpack/rfcs/blob/project-descriptor-simple/text/0000-project-descriptor.md#example-custom-image-prefix) as discussed.

jkutner commented:
@sclevine @nebhale @hone @ekcasey ready for re-review

jkutner commented:
I've moved the metadata keys in to `[project]` based on the precedent found in `pom.xml` and `package.json`.

I've added `ref` and `suffix` keys to `[[images]]` as described in WG. I chose `ref` because I was afraid `tag` too strongly referred to a complete tag. Happy to change that though.

sclevine commented on text/0000-project-descriptor.md:
```diff
@@ -0,0 +1,324 @@
+exclude = ["<string>"]
+
+[[images]]
+ref = "<string>"
```
Small issue with `ref` vs `tag`: only tags are valid in this field, and other kinds of ref are not valid.
E.g., `gcr.io/example/myapp@sha256:2342ab...` is not valid, because it's not possible to know the digest ahead of time.

jkutner replied:
i'm fine with `tag` then

sclevine commented on text/0000-project-descriptor.md:
````diff
@@ -0,0 +1,324 @@
+
+```toml
+[project]
+id = "<string>"
````
Were we going to replace `id`/`version` with `ref` (or `tag`, noting my comment below), and then combine it with the individual `ref`s and `suffix`s?

jkutner replied:
i think i'm fine with that. i'd probably move to make it optional though (which means there's no required id or tag, which might be a problem).

sclevine commented on text/0000-project-descriptor.md:
````diff
@@ -0,0 +1,324 @@
+```toml
+[project]
+id = "io.buildpacks.my-app"
+version = "0.1"
````
Seems unusual to require a version to be checked into a git repo for project.toml to be valid. That breaks certain release strategies.

sclevine commented on text/0000-project-descriptor.md:
```diff
@@ -0,0 +1,324 @@
+
+## Example: Monorepos
+
+The `[[images]]` array of tables allows the buildpack lifecycle to generate more than one image per build. Each table in the array may contain configuration that defines a different image for different parts of the code. For example:
```
clarify per `pack build`, not lifecycle build

hone commented:
From @jonjohnsonjr, Julia also has a [Project.toml](https://github.com/JuliaLang/Pkg.jl/issues/805).

jkutner commented:
Closing in favor of #32. We'll open a new RFC for `[[images]]`, `[[extensions]]`, buildpacks and other things.
